server/src/main/java/io/crate/planner/selectivity/SelectivityFunctions.java

```diff
@@ -129,12 +142,70 @@ public Double visitFunction(Function function, Void context) {
                 }
 
                 default:
+                    if (Operators.COMPARISON_OPERATORS.contains(function.name())) {
+                        return genericOpSelectivity(nodeCtx, txnCtx, stats, function, params);
+                    }
                     return MAGIC_SEL;
             }
         }
 
     }
 
+    @SuppressWarnings("unchecked")
+    private static double genericOpSelectivity(NodeContext nodeCtx,
+                                               TransactionContext txnCtx,
+                                               Stats stats,
+                                               Function function,
+                                               Row params) {
+        assert Operators.COMPARISON_OPERATORS.contains(function.name())
+            : "genericOpSelectivity only works for Operators like >, <, <= and >=";
+
+
+        double defaultSel = MAGIC_SEL;
+
+        List<Symbol> arguments = function.arguments();
+        if (arguments.size() != 2) {
```

Comment:
Is this a valid case? shouldn't we have an assertions instead?

Reply:
Changed it to an assert